app/internal/import_file.py

```diff
@@ -0,0 +1,193 @@
+                same_date_counter = date_n_count[event["S_Date"]]
+        else:
+            date_n_count[event["S_Date"]] = 1
+    return same_date_counter <= max_event_start_date
+
+
+def import_txt_file(txt_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(txt_file, "r") as text:
+        events = text.readlines()
+    for event in events:
+        if not is_event_text_valid(event):
+            return list()
+        head, content, start_date, end_date = event.split(", ")
+        if not is_date_in_range(start_date) or \
+           not is_date_in_range(end_date.replace("\n", "")):
+            return list()
+        start_date = datetime.datetime.strptime(start_date, "%m-%d-%Y")
+        end_date = datetime.datetime.strptime(end_date.replace("\n", ""),
+                                              "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "S_Date": start_date,
+                                 "E_Date": end_date})
+    return calendar_content
+
+
+def open_ics(ics_file: str):
+    with open(ics_file, "r") as ics:
+        try:
+            calendar_read = Calendar.from_ical(ics.read())
+        except (IndexError, ValueError):
+            return list()
+    return calendar_read
+
+
+def is_valid_data_event_ics(component) -> bool:
+    """check if ics event data content is valid"""
+    if str(component.get('summary')) is None or \
+       component.get('dtstart') is None or \
+       component.get('dtend') is None or \
+       not is_date_in_range(component.get('dtstart').dt) or \
+       not is_date_in_range(component.get('dtend').dt):
+        return False
+    return True
+
+
+def save_calendar_content_ics(component, calendar_content) -> None:
+    calendar_content.append({
+                        "Head": str(component.get('summary')),
+                        "Content": str(component.get('description')),
+                        "S_Date": component.get('dtstart').dt
+                        .replace(tzinfo=None),
+                        "E_Date": component.get('dtend').dt
+                        .replace(tzinfo=None)
+                    })
+
+
+def import_ics_file(ics_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    calendar_read = open_ics(ics_file)
+    if not calendar_read:
+        return list()
+    for component in calendar_read.walk():
+        if component.name == "VEVENT":
+            if is_valid_data_event_ics(component):
+                save_calendar_content_ics(component, calendar_content)
+            else:
+                return list()
+    return calendar_content
+
+
+def save_events_to_database(events: List[Dict[str, Union[str, Any]]],
+                            user_id: int,
+                            session: SessionLocal) -> None:
+    """insert the events into Event table"""
+    for event in events:
+        event = Event(
```

Comment:
What about the event's location?

Reply:
The code is constantly changing,
I will update this with a new ticket after I characterize all the new changes required and tests
In order to maintain quality follow-up :)